### ISSUE TITLE
French SI Brochure feedback from BIPM (pages 21-40)

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/bipm-si-brochure#41]
page | comment
-- | --
21 | first equation: wrong grouping of decimals after the comma: 6,022 140 76.
21 | second paragraph after 1. equation: "M" needs always to be initalics
21 | 2. equation: the 12 in 12C needs to be a superscript, not a subscript; C not in italics
21 | below 3. equation: an opening parenthesis is missing in M(12C)
22 | definition of candela: 2nd last line it should read m-2,not m2; Negative and positive exponents are not at the same height
22 | 2. equation: replace decimal point by comma; insert a space between h and Kcd (2 x)
23 | 3. paragraph: replace "Article 5.4.7" with "section 5.4.7" and analogous at several other places
24 | first paragraph of 2.3.4: isolated » on new line
24 | Table 4: angle solide: m2 not in italics
24 | Table 4: 3re column vertically not well aligned with 1st and 2nd column
24 | Table 4, footnote b: pi not in italics
27 | Table 6: energie volumique: last column: kg m-1 s-2; there is a misprint in the SI brochure
27 | Table 6: increase width of first two columns so that there are less linebreaks
28 | 4. paragraph: pi not in italics
28 | 4. paragraph : isolated » on new line
28 | 5. paragraph : isolated » on new line
30 | side note: accents missing on four prefix names: mébi, tébi, zébi,pébi
30 | μ not in italics
31 | last line : μ not in italics
32 | Table 8: pi not in italics
32 | Table 8 footnot b: μ not in italics
33 | 1. paragraph: 3rd line from bottom: m in italics
34 | 3. paragraph: split in two paragraphs before "Les symboles…"
35 | section 5.4.1, 2. paragraph: "livre rouge" not in italics ;  isolated » on new line
36 | Figure: insert ticks for the axis labels
36 | sectio 5.4.2, last side note: spaces not correct, compare with original
37 | 2. line: reference to page number 149 wrong, should be 140
37 | 5.4.4: isolated » on new line
38 | 5.4.7, 2. paragraph: μ not in italics
38 | 5.4.7, 3. paragraph: isolate « at end of line
38 | last side note,last line: space missing between" confusion" and "supplémentaire"
39 | 5.4.8: pi not in italics
39 | 4. paragraph: rayon r, r in italics; omega in italics
40 | Full name of CIPM and CGPM instead of acronym
40 | 1. paragraph: "procès verbaux" in italics

**Comment by Intelligent2013** [comment]
page | comment | to fix
-- | -- | --
21 | first equation: wrong grouping of decimals after the comma: 6,022 140 76. | adoc
21 | second paragraph after 1. equation: "M" needs always to be initalics | adoc
21 | 2. equation: the 12 in 12C needs to be a superscript, not a subscript; C not in italics | adoc
21 | below 3. equation: an opening parenthesis is missing in M(12C) | adoc
22 | definition of candela: 2nd last line it should read m-2,not m2; Negative and positive exponents are not at the same height | adoc, jEuclid?
22 | 2. equation: replace decimal point by comma; insert a space between h and Kcd (2 x) | adoc
23 | 3. paragraph: replace "Article 5.4.7" with "section 5.4.7" and analogous at several other places | rb
24 | first paragraph of 2.3.4: isolated » on new line | adoc
24 | Table 4: angle solide: m2 not in italics | adoc
24 | Table 4: 3re column vertically not well aligned with 1st and 2nd column | xslt
24 | Table 4, footnote b: pi not in italics | adoc
27 | Table 6: energie volumique: last column: kg m-1 s-2; there is a misprint in the SI brochure | adoc
27 | Table 6: increase width of first two columns so that there are less linebreaks | ?
28 | 4. paragraph: pi not in italics | adoc
28 | 4. paragraph : isolated » on new line | adoc
28 | 5. paragraph : isolated » on new line | adoc
30 | side note: accents missing on four prefix names: mébi, tébi, zébi,pébi | adoc
30 | μ not in italics | adoc
31 | last line : μ not in italics | adoc
32 | Table 8: pi not in italics | adoc
32 | Table 8 footnot b: μ not in italics | adoc
33 | 1. paragraph: 3rd line from bottom: m in italics | adoc
34 | 3. paragraph: split in two paragraphs before "Les symboles…" | adoc
35 | section 5.4.1, 2. paragraph: "livre rouge" not in italics ;  isolated » on new line | adoc
36 | Figure: insert ticks for the axis labels | image
36 | sectio 5.4.2, last side note: spaces not correct, compare with original | adoc
37 | 2. line: reference to page number 149 wrong, should be 140 | adoc
37 | 5.4.4: isolated » on new line | adoc
38 | 5.4.7, 2. paragraph: μ not in italics | adoc
38 | 5.4.7, 3. paragraph: isolate « at end of line | adoc
38 | last side note,last line: space missing between" confusion" and "supplémentaire" | adoc
39 | 5.4.8: pi not in italics | adoc
39 | 4. paragraph: rayon r, r in italics; omega in italics | adoc
40 | Full name of CIPM and CGPM instead of acronym | adoc
40 | 1. paragraph: "procès verbaux" in italics | adoc

**Comment by opoudjis** [comment]
For rb, see #40

**Comment by manuel489** [comment]
>21 | first equation: wrong grouping of decimals after the comma: 6,022 140 76. | adoc

I understand that this is suposed to be handled automatically (and not by markup).


> 21 | second paragraph after 1. equation: "M" needs always to be initalics | adoc

>  21 | 2. equation: the 12 in 12C needs to be a superscript, not a subscript; C not in italics | adoc

> 21 | below 3. equation: an opening parenthesis is missing in M(12C) | adoc

All three fixed.

> 22 | definition of candela: 2nd last line it should read m-2,not m2; Negative and positive exponents are not at the same height | adoc, jEuclid?

This is a rendering issue. Not fixable by markup.

> 22 | 2. equation: replace decimal point by comma; insert a space between h and Kcd (2 x) | adoc

Fixed

> 24 | first paragraph of 2.3.4: isolated » on new line | adoc

Ok, I believe that `« »` are quotation marks, right? So, in order to avoid the issue, I've removed the white space between quotation marks and their adjacent words.

> 24 | Table 4: angle solide: m2 not in italics | adoc

Fixed.

> 24 | Table 4, footnote b: pi not in italics | adoc

This is not fixable by markup. @opoudjis?

> 27 | Table 6: energie volumique: last column: kg m-1 s-2; there is a misprint in the SI brochure | adoc

This a rendering isssue. Not fixable by markup.

> 27 | Table 6: increase width of first two columns so that there are less linebreaks | ?

I can change attributes of the table in order to indicate that certain columns must be wider than others (e.g. `[cols="2,2,1]`). But this seems not to have any influence in the output.

> 28 | 4. paragraph: pi not in italics | adoc

Not fixable by markup

> 28 | 4. paragraph : isolated » on new line | adoc

> 28 | 5. paragraph : isolated » on new line | adoc

White space removed. (See comment above.)

> 30 | side note: accents missing on four prefix names: mébi, tébi, zébi,pébi | adoc

Fixed.

> 30 | μ not in italics | adoc

> 31 | last line : μ not in italics | adoc

> 32 | Table 8: pi not in italics | adoc

> 32 | Table 8 footnot b: μ not in italics | adoc

Not fixable by markup (unless I pasted the literal symbol). @ronaldtse ?

> 33 | 1. paragraph: 3rd line from bottom: m in italics | adoc

> 34 | 3. paragraph: split in two paragraphs before "Les symboles…" | adoc

> 35 | section 5.4.1, 2. paragraph: "livre rouge" not in italics ; isolated » on new line | adoc

All three fixed.

> 36 | Figure: insert ticks for the axis labels | image

I changed the image file.

> 36 | sectio 5.4.2, last side note: spaces not correct, compare with original | adoc

Fixed. Line breaks were inserted manually.

> 37 | 2. line: reference to page number 149 wrong, should be 140 | adoc

Not sure what to do here. I changed "149" by "140", but should I put a cross-reference here? If so, this cross reference must link to an external document (which is the English version of Brochure). @opoudjis?

> 37 | 5.4.4: isolated » on new line | adoc

White space removed.

> 38 | 5.4.7, 2. paragraph: μ not in italics | adoc

Not fixable by markup (unless I pasted the literal symbol).

> 38 | 5.4.7, 3. paragraph: isolate « at end of line | adoc

White space removed.

> 38 | last side note,last line: space missing between" confusion" and "supplémentaire" | adoc

Fixed.

> 39 | 5.4.8: pi not in italics | adoc

Not fixable by markup (unless I pasted the literal symbol).

> 39 | 4. paragraph: rayon r, r in italics; omega in italics | adoc

Fixed "rayon r" part. But "omega in italics" is not fixable by markup.

> 40 | Full name of CIPM and CGPM instead of acronym | adoc

> 40 | 1. paragraph: "procès verbaux" in italics | adoc

All two fixed.

**Comment by manuel489** [comment]
The fixes are in PR https://github.com/metanorma/bipm-si-brochure/pull/45

**Comment by Intelligent2013** [comment]
Regarding ` isolated » on new line` issues. 
I think we should not remove space between `double angle quotation` and text.

Source renders with spaces:
![изображение](https://user-images.githubusercontent.com/5559926/97787071-ea175300-1bc0-11eb-8716-67cdec6a8128.png)

I think that would be bettet to replace space to non-break space character (hex A0).

**Comment by manuel489** [comment]
Thanks @Intelligent2013! I take your suggestion. Changes applied.

**Comment by opoudjis** [comment]
@manuel489 

> 37 | 2. line: reference to page number 149 wrong, should be 140 | adoc
> Not sure what to do here. I changed "149" by "140", but should I put a cross-reference here? If so, this cross reference must link to an external document (the English version of Brochure). @opoudjis?

That's a nightmare scenario, but we do provide for it under collections. The problem is that this is going to force a massive unwelcome change to how we process crossreferences between documents.

To reference another document in a collection, use:

`* [[[english-doc,repo:(current-metanorma-collection/si-brochure-en)]]]`

and 

`<<english-doc,anchor=id>>`

I think this will break. Do it anyway. We will have to deal with it when it does.

**Comment by opoudjis** [comment]
> 24 | Table 4, footnote b: pi not in italics | adoc
> This is not fixable by markup. @opoudjis?

@Intelligent2013 bringing you in here.

The XML here is:

`<stem type="MathML"><math xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>π</mi><mtext> rad</mtext></math></stem>`

Now, the MathML spec explicitly says that Greek constants like pi must be marked up as mi, not mn: https://www.w3.org/TR/MathML3/chapter3.html#presm.mi . The problem is, MathJax, and clearly Euclid, are blanket rendering anything in mi as italics, and Greek lowercase letters are not italicised in maths. (Practice varies for capital Greek letters, and MathJax treats it as an amsmath config option: https://tex.stackexchange.com/questions/87238/greek-letters-in-italic-in-math-equation.)

As https://tex.stackexchange.com/a/440589 indicates, this is a local preference:

> With a modern toolchain that supports unicode-math, you can select math-style=upright to get all upright letters, and math-style=ISO to get all slanted math letters. Upright numbers and Greek capital letters with all other letters in math mode italicized is the default, but you can select it explicitly with the option math-style=TeX. There is also a fourth math-style=French option. Whatever you select, symbols such as summation and product series remain upright.

And sure enough:

https://tex.stackexchange.com/questions/159973/french-math-style-with-default-font

> The package mathastext called with the frenchmath option does the job.

> I've realized that French typography uses a different style for mathematics, namely upright uppercase roman letters and lowercase greek letters (roman lowercase remaining in italic), while default LaTeX is italic. What I want is to use this style with the default math font.

This is a rendering option in TeX. We need to find out if this is a rendering option in jEuclid and in MathJax.

This will NOT be addressed by manually injecting italics or CSS in MathML. We might get away with that, just, in Word; we will not get away with that in PDF.

The BIPM HTML, risibly, used GIFs for Greek letters. No, I am not making this up: https://www.bipm.org/fr/CGPM/db/26/1/ . They therefore have no ability whatsoever to provide any guidance for the proper HTML rendering of MathML, and little authority to demand French rendering.

**Comment by opoudjis** [comment]
Flabbergastingly, the MathJax TeX font, which is the default font MathJax uses, has no support for upright lowercase Greek characters at all: a non-italic pi is impossible within MathJax even with the "mathvariant" override attribute (which I really do not want to use). We would have to shift to STIX-Web font to display any upright Greek fonts at all.

**Comment by opoudjis** [comment]
mathvariant is working in HTML if I change the MathJax font to STIX. @Intelligent2013 is investigating whether it is respected in jEuclid. If it is, I will need to postprocess the MathML to make it behave as expected in French mathematical typesetting. If it is not, our answer has to be that we do not support French mathematical typesetting.

**Comment by Intelligent2013** [comment]
I've checked `mathvariant` attribute and it is working in jEuclid, here is resulted PDF:
![изображение](https://user-images.githubusercontent.com/5559926/97801287-a2d6a400-1c4c-11eb-83df-e694a873378c.png)

jEuclid set mathvariant style for `mi` tag (from jeuclid-core\src\main\java\net\sourceforge\jeuclid\elements\presentation\token\Mi.java) depends on text length:
```java
        if (StringUtil.countDisplayableCharacters(this.getText()) == 1) {
            this.setDefaultMathAttribute(
                    AbstractJEuclidElement.ATTR_MATHVARIANT, "italic");
        } else {
            this.setDefaultMathAttribute(
                    AbstractJEuclidElement.ATTR_MATHVARIANT, "normal");
        }
```

Regarding https://www.w3.org/TR/MathML3/chapter3.html#presm.mi, the default value of `mathvariant` depends of length of content:
![изображение](https://user-images.githubusercontent.com/5559926/97801329-0cef4900-1c4d-11eb-93e2-cbcddeb518ca.png)

If we append zero-width space after pi inside `mi`:
```xml
<stem type="MathML"><math xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>π​</mi><mtext> rad</mtext></math></stem>
```

then we get non-italic pi:
![изображение](https://user-images.githubusercontent.com/5559926/97801528-70c64180-1c4e-11eb-86ef-d98c31bb9b15.png)

**Comment by opoudjis** [comment]
Ok. This becomes new ticket: https://github.com/metanorma/metanorma-standoc/issues/367

**Comment by Intelligent2013** [comment]
Regarding issue:
page | comment
-- | --
22 | ... Negative and positive exponents are not at the same height

![изображение](https://user-images.githubusercontent.com/5559926/97804173-226d6e80-1c5f-11eb-9078-f4932c6e0edf.png)

The reason of misaligment between `-2` and `-1` is the max height of previous chars - `k` has a more height than other chars `g s r`, therefore `-1` shows a bit higher.
I've checked how it displays in MathJax (https://www.mathjax.org/#demo), and it shows similar (but with less differences):

![изображение](https://user-images.githubusercontent.com/5559926/97804212-75dfbc80-1c5f-11eb-88f8-bfd3aab2e7d2.png)


I see only one solution - isolating `cd sr k` from `g` into separated mtext like this:
```<mtext> cd sr k</mtext>```

I,.e. from:
```
<msup><mrow><mtext> cd sr kg</mtext></mrow><mrow><mrow><mo>−</mo><mn>1</mn></mrow></mrow></msup>
```

to
```
<mtext> cd sr k</mtext><msup><mrow><mtext>g</mtext></mrow><mrow><mrow><mo>−</mo><mn>1</mn></mrow></mrow></msup>
```

Can we do it? If so, then we'll get this result:
![изображение](https://user-images.githubusercontent.com/5559926/97804535-aaed0e80-1c61-11eb-80b8-73bfac68e8fd.png)

**Comment by opoudjis** [comment]
Um...

... That solution does such violence to the MathML, I would much rather we refuse to implement it at all. Whatever @ronaldtse has planned for UnitsML, it will decidedly not permit this kind of rearrangement of text: units will be treated as semantically meaningful elements.

**Comment by opoudjis** [comment]
This *could* be addressed by adding an `msup/@superscriptshift` attribute, forcing the same minimum height on all superscripts. But good luck in guessing what that might be.

So this does work:

`<math><mrow><mtext>cd sr </mtext><msup superscriptshift="4pt"><mrow><mi>kg</mi></mrow><mrow><mo>−</mo><mn>1</mn></mrow></msup><msup superscriptshift="4pt"><mi mathvariant="normal">m</mi><mrow><mo>−</mo><mn>2</mn></mrow></msup></mrow></math>`

But... how do I know what the right point measure for superscripting should be? We really should not be being forced to do this amount of fine tuning of rendering.

**Comment by manuel489** [comment]
> That's a nightmare scenario, but we do provide for it under collections. The problem is that this is going to force a massive unwelcome change to how we process crossreferences between documents.
> 
> To reference another document in a collection, use:
> 
> `* [[[english-doc,repo:(current-metanorma-collection/si-brochure-en)]]]`
> 
> in the bibliography, and
> 
> `<<english-doc,anchor=id>>`
> 
> I think this will break. Do it anyway. We will have to deal with it when it does.

@opoudjis , problem is, there is no bibliography section in original document. I would need to create one.  Should I do so?

**Comment by Intelligent2013** [comment]
Issue:
page | comment
-- | --
22 | ... Negative and positive exponents are not at the same height

fixed.

Before:
![изображение](https://user-images.githubusercontent.com/5559926/97804173-226d6e80-1c5f-11eb-9078-f4932c6e0edf.png)

After:
![изображение](https://user-images.githubusercontent.com/5559926/98466381-33c6f580-21e0-11eb-88e0-915325b9cd3b.png)

**Comment by opoudjis** [comment]
> @opoudjis , problem is, there is no bibliography section in original document. I would need to create one. Should I do so?

Do so for now. I will have to implement code to remove it from ultimate rendering, and this is an issue for ISO 10303 as well, but the reference needs to be in there before we can process the reference at all.

**Comment by manuel489** [comment]
> > @opoudjis , problem is, there is no bibliography section in original document. I would need to create one. Should I do so?
> 
> Do so for now. I will have to implement code to remove it from ultimate rendering, and this is an issue for ISO 10303 as well, but the reference needs to be in there before we can process the reference at all.

Changes pushed to master branch directly.

**Comment by opoudjis** [comment]
https://github.com/metanorma/metanorma/issues/146 will address removing (hiding) those internal bibliographic entries.

**Comment by manuel489** [comment]
Based on the latest PDF provided by @opoudjis: [si-brochure-fr(12-01).pdf](https://github.com/metanorma/bipm-si-brochure/files/5622920/si-brochure-fr.12-01.pdf)

These are the issues that don't fulfill with BIPM requirements yet.

page | comment | issue
-- | -- | --
21 | second paragraph after 1. equation: "M" needs always to be in italics | not fixed
22 | definition of candela: 2nd last line it should read m-2,not m2; Negative and positive exponents are not at the same height | not fixed
27 | Table 6: increase width of first two columns so that there are less linebreaks | not fixed
36 | sectio 5.4.2, last side note: spaces not correct, compare with original | fixed in https://github.com/metanorma/bipm-si-brochure/pull/59
37 | 2. line: reference to page number 149 wrong, should be 140 |  not fixed
39 | 4. paragraph: rayon r, r in italics; omega in italics | omega part not fixed

**Comment by Intelligent2013** [comment]
>  22 | definition of candela: 2nd last line it should read m-2,not m2; Negative and positive exponents are not at the same height

Fixed. I've increased a bit the height of exponents.

Before:
![изображение](https://user-images.githubusercontent.com/5559926/101168429-7e575880-364c-11eb-9921-f77336495150.png)

After:
![изображение](https://user-images.githubusercontent.com/5559926/101168297-4b14c980-364c-11eb-8c9b-fb01cc8b104b.png)

> 27 | Table 6: increase width of first two columns so that there are less linebreaks

I can't change column width for concrete table. Columns width set automatically depends on longest word in cell. For this table:
- 1st column - width 13 (concentration)
- 2nd column - width 10 (kilogramme)
- 3rd column - width 11 (
![изображение](https://user-images.githubusercontent.com/5559926/101169453-fd995c00-364d-11eb-9767-333dc0b3f463.png))

- 4nd column - width 19 (
![изображение](https://user-images.githubusercontent.com/5559926/101169480-0ab64b00-364e-11eb-8c8f-8503e8a3338d.png))

As workaround solution you can try to split long stems in last column into two stems.


>  37 | 2. line: reference to page number 149 wrong, should be 140

This page number shows only in two-languages PDF. In French only it doesn't show.

**Comment by opoudjis** [comment]
> 39 | 4. paragraph: rayon r, r in italics; omega in italics | omega part not fixed

See https://github.com/metanorma/bipm-si-brochure/issues/40#issuecomment-738837545

> 21 | second paragraph after 1. equation: "M" needs always to be in italics | not fixed

Metanorma is implementing the French mathematical typesetting rule, that uppercase variables are upright. Where that is ignored, you need to introduce explicit font styling: `stem:[ii(M)_{rm(u)}]`.

Note that this won't work until https://github.com/metanorma/metanorma-standoc/pull/386 is merged in; it isn't yet, because of a bug somewhere else in the stack.

**Comment by manuel489** [comment]
One remaining issues in this thread:

page | comment | issue
-- | -- | --
21 | second paragraph after 1. equation: "M" needs always to be in italics | fixed
22 | definition of candela: 2nd last line it should read m-2,not m2; Negative and positive exponents are not at the same height | fixed, https://github.com/metanorma/bipm-si-brochure/issues/41#issuecomment-738786469
27 | Table 6: increase width of first two columns so that there are less linebreaks | not fixed
36 | sectio 5.4.2, last side note: spaces not correct, compare with original | fixed
37 | 2. line: reference to page number 149 wrong, should be 140 |  will not fix, https://github.com/metanorma/bipm-si-brochure/issues/41#issuecomment-738786469
39 | 4. paragraph: rayon r, r in italics; omega in italics | fixed

**Comment by manuel489** [comment]
>  27 | Table 6: increase width of first two columns so that there are less linebreaks | not fixed

@opoudjis it seems that is not possible to increase the column width in tables by using AsciiDoc attributes like:
 `[cols="2,2,1,1"]` (a four column table where the first two columns are wider than the last ones.)

Is there another way to address this issue?

**Comment by opoudjis** [comment]
There is a ticket requesting that I introduce column width: https://github.com/metanorma/metanorma-standoc/issues/251

There is no other way to address it, but that would. I don't like doing it, being a purist, but I don't think I can avoid this in the long term. WIll realise that ticket.

**Comment by anermina** [comment]
Since the new ticket is opened to address this issue, I'm closing this one. I'll create a new ticket for these remaining issues which require more time to be solved, so that we can track what's left for the BIPM.
